### PR TITLE
Refactor tensor initialization check for external weights and fixed lint issues

### PR DESCRIPTION
**Pull request by jatinwadhwa921**
Refactor tensor initialization check for external weights and fixed lint issues

### CODE REVIEW

File: onnxruntime/core/providers/openvino/backends/basic_backend.cc [hunk 1]
```diff
@@ -81,7 +81,7 @@ BasicBackend::BasicBackend(std::unique_ptr<ONNX_NAMESPACE::ModelProto>& model_pr
       }
 #else  // !IO_BUFFER_ENABLED
       std::string prec_str = (global_context_.precision_str != "ACCURACY") ? global_context_.precision_str : global_context_.model_precision;
-      if (is_ep_ctx_graph_) {
+      if (is_ep_ctx_graph_ && !global_context_.has_external_weights) {
```
[the remark this answers is not in the document]
Comment (jatinwadhwa921) [reply onnxruntime/core/providers/openvino/backends/basic_backend.cc]: yes, you are correct, made the changes and tested them as well

**Comment by jatinwadhwa921** [comment]
> Passing this variable across call stack might be difficult to trace in long term. Can you explore the ng_required_initialisers in capability.cc or data_ops.cc

External weights can be associated with any initializer in the ONNX model, so we can miss the initializer if we are just traversing the nodes which are not supported as ng_required_initilisers points to the nodes which are not supported, so i believe we should go ahead with my method for now

**Comment by preetha-intel** [comment]
LGTM

**Comment by preetha-intel** [comment]
@jatinwadhwa921 Rebase with latest changes.

**Comment by preetha-intel** [comment]
PR #464 is merged . Closing this as its no longer relevant.

**Pull request closed without merging**